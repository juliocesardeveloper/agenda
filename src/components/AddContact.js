import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddContact = ({ setContacts }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ( e ) => {
    setInputValue( e.target.value );
  }

  const handleSubmit = ( e ) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setContacts( cntcts => [ inputValue, ...cntcts ] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <input
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}


AddContact.propTypes = {
  setContacts: PropTypes.func.isRequired
}
