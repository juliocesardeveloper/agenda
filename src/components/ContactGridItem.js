import React from 'react'

export const ContactGridItem = ({ avatar, name, email, phone }) => {

  return (
    <div className="card">
      <img src={ avatar } alt={ name }/>
      <h2> { name } </h2>
      <p> <b>Email:</b> { email } </p>
      <p> <b>Phone:</b> { phone } </p>
    </div>
  )
}
