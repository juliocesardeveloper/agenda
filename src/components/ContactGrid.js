import React from 'react'
import { useFetchContacts } from '../hooks/useFetchContacts'
import { ContactGridItem } from './ContactGridItem'

export const ContactGrid = ({ contact }) => {

  const { data:names, loading } = useFetchContacts( contact );
  const allContacts = names;
  

  return (
    <>
      <div className="card-grid-titles animate__animated animate__fadeIn">
        {
          contact ? <h2>Resultados de BÃºsqueda: { contact }</h2> : <h2 className="contactos">Contactos</h2>
        }
      </div>

      { loading && <p className="animate__animated animate__flash">Cargando</p> }

      <div className="card-grid" >
        {
          allContacts.map( name => (
            <ContactGridItem
              key={ name.id }
              { ...name }
            />
          ) )
        }
      </div>
    </>
  )
}
