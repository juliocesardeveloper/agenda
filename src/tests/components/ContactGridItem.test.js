import React from 'react'
import { shallow } from 'enzyme'
import { ContactGridItem } from '../../components/ContactGridItem'

describe('Pruebas en ContactGridItem', () => {
  test('Debe mostrar el componente correctamente', () => {
    const wrapper = shallow( <ContactGridItem /> )
    expect( wrapper ).toMatchSnapshot();
  })
})