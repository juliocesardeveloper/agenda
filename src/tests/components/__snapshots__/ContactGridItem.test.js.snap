// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ContactGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <h2>
     
     
  </h2>
  <p>
     
    <b>
      Email:
    </b>
     
     
  </p>
  <p>
     
    <b>
      Phone:
    </b>
     
     
  </p>
</div>
`;
